contracts/adapters/CharmAdapter.sol

```diff
@@ -0,0 +1,381 @@
+    }
+
+    /**
+     * @notice Swaps the exercised profit (originally in the collateral token) into the `underlying` token.
+     *         This simplifies the payout of an option. Put options pay out in USDC, so we swap USDC back
+     *         into WETH and transfer it to the recipient.
+     * @param baseToken is the base token of the market
+     * @param profitInBaseToken is the profit after exercising denominated in the base token - this could be a token with different decimals
+     * @param recipient is the recipient of the underlying tokens after the swap
+     */
+    function _swapExercisedProfitsToUnderlying(
+        IERC20 baseToken,
+        uint256 profitInBaseToken,
+        address recipient
+    ) private returns (uint256 profitInUnderlying) {
+
+        require(
+            baseToken.uniBalanceOf(address(this)) >= profitInBaseToken,
+            "Not enough collateral from exercising"
+        );
+
+        IUniswapV2Router02 router = IUniswapV2Router02(UNISWAP_ROUTER);
+
+        if (address(baseToken) == address(0)) {
+            (bool success, ) = recipient.call{value: profitInBaseToken}("");
+            require(success, "Failed to transfer exercise profit");
+        } else {
+            address[] memory path = new address[](2);
+            path[0] = address(baseToken);
+            path[1] = address(WETH);
+
+            uint256[] memory amountsOut =
+                router.getAmountsOut(profitInBaseToken, path);
+            profitInUnderlying = amountsOut[1];
+
+            require(profitInUnderlying > 0, "Swap is unprofitable");
+
+            baseToken.safeApprove(address(router), profitInBaseToken);
+
+            router.swapExactTokensForETH(
+                profitInBaseToken,
+                profitInUnderlying,
+                path,
+                recipient,
+                block.timestamp + SWAP_WINDOW
+            );
+        }
+      }
+
+      /**
+       * @notice Swaps the ETH into the `base` token.
+       *         This simplifies the buying of an option since you only pay in ETH for any option
+       * @param baseToken is the base token of the market
+       * @param maxCost max cost we want to spend
+       * @param premium premium to buy option
+       * @param market market of token
+       */
+       function _swapETHToBaseToken(
+           IERC20 baseToken,
+           uint256 maxCost,
+           uint256 premium,
+           IOptionMarket market
+       ) private{
+        //uint256 premium = optionViews.getBuyOptionCost(IOptionMarket(market), isLong, strikeIndex, shiftedAmount);
+
+        IUniswapV2Router02 router = IUniswapV2Router02(UNISWAP_ROUTER);
+
+        address[] memory path = new address[](2);
+        path[0] = WETH;
+        path[1] = address(baseToken);
+
+        router.swapETHForExactTokens{value: address(this).balance}(
+            premium,
```

Comment:
We can probably just use the `maxCost` instead of `premium` for this argument. This avoids us calling `optionViews.getBuyOptionCost` in the first place

Reply:
so you are saying you want to pre-compute the premium offchain and pass it in as maxCost?